assets/js/base/components/cart-checkout/product-details/index.tsx

```diff
@@ -30,20 +30,22 @@ const ProductDetails = ( {
 	return (
 		<ul className="wc-block-components-product-details">
 			{ details.map( ( detail ) => {
-				const className = detail.name
+				// Support both `key` and `name` props
+				detail.key = detail.key ? detail.key : detail.name || '';
+				const className = detail.key
 					? `wc-block-components-product-details__${ kebabCase(
-							detail.name
+							detail.key
 					  ) }`
 					: '';
 				return (
 					<li
-						key={ detail.name + ( detail.display || detail.value ) }
+						key={ detail.key + ( detail.display || detail.value ) }
 						className={ className }
 					>
-						{ detail.name && (
+						{ detail.key && (
 							<>
 								<span className="wc-block-components-product-details__name">
-									{ decodeEntities( detail.name ) }:
+									{ decodeEntities( detail.key ) }:
```

Comment:
Just pointing out, if `name` makes more sense in this implementation (looking at the CSS classname), we could still respect `key` e.g. `const name = detail?.key || detail.name || '';`

Reply:
@sunyatasattva @mikejolley So based on the above discussion, I've updated the PR with @mikejolley's suggestion here, which removes overwriting the object key, renames the const to `name` for better readability in the code here, and respects the intent of using the name if the key is empty.